app/controllers/admin_controller.rb

```diff
@@ -93,12 +97,38 @@ def show_instructor
       @role = Role.new(:id => nil, :name => '(none)')
     end
   end
+  
+  def show_super_admin
+    @user = User.find(params[:id])
+    if @user.role_id
+      @role = Role.find(@user.role_id)
+    else
+      @role = Role.new(:id => nil, :name => '(none)')
+    end
+  end
+  def show_admin
+    @user = User.find(params[:id])
+    if @user.role_id
+      @role = Role.find(@user.role_id)
+    else
+      @role = Role.new(:id => nil, :name => '(none)')
+    end
+  end
 
   def remove_instructor
     User.find(params[:id]).destroy
     redirect_to :action => 'list_instructors'
   end
 
+  def remove_administrator
```

Comment:
The method remove_administrator and remove_super_administrator look scary to me.